src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5094,8 +5059,47 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         )
         silentResult match {
           case SilentResultValue(fun1) =>
+            def transformRassoc(applied: Tree): Tree = {
+              val Apply(fn, arg :: Nil) = applied: @unchecked
+              val isByName = fn.symbol.paramLists.head.head.isByNameParam
+              if (isByName) applied
+              else {
+                def isRassocArg(t: Tree) = t.hasAttachment[RightAssociativeArg.type]
+                def checkRassocArg(t: Tree): Boolean = isRassocArg(t) && { t.removeAttachment[RightAssociativeArg.type]; true }
+                def rewriteRightAssoc(arg: Tree): Tree = {
+                  val vsym = context.owner.newValue(freshTermName(nme.RIGHT_ASSOC_OP_PREFIX), arg.pos.focus, FINAL | SYNTHETIC | ARTIFACT)
+                  vsym.setInfo(arg.tpe)
+                  val vdef = atPos(arg.pos) { ValDef(vsym, arg) setType NoType }
+                  context.pendingStabilizers ::= vdef
+                  arg.changeOwner(context.owner -> vsym)
+                  Ident(vsym).setType(singleType(NoPrefix, vsym)).setPos(arg.pos.focus)
+                }
+                def needsRewrite(t: Tree) = !treeInfo.isStableIdentifier(t, allowVolatile = false) && !treeInfo.isExprSafeToInline(t)
+                val rewriter = new Transformer {
+                  override def transform(tree: Tree): Tree = tree match {
+                    //case TypeApply(fn, _) if checkRassocArg(fn) =>
+                    //  if (needsRewrite(fn)) rewriteRightAssoc(tree) else tree
+                    case Apply(fn, Apply(arg, Nil) :: Nil) if checkRassocArg(arg) =>
+                      // tree is adapted empty application arg(), so move the adapted tree
+                      if (needsRewrite(tree)) rewriteRightAssoc(tree) else tree
```

Comment:
I'm a bit confused here, why it calls `rewriteRightAssoc(tree)` not `rewriteRightAssoc(the-argument)`? Also this case seems ad-hoc, there could be other adaptations?